[Regression] Prisma 6.11.0 - prisma+postgres is not a supported database URL

### Bug description

New version breaks the db connection via Prisma Accelerate:

```
PrismaClientInitializationError: The provided database string is invalid. Error parsing connection string: prisma+postgres is not a supported database URL scheme. in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
    at r (/Users/daviddakhovich/Documents/Projects/Dourjob/backend/node_modules/@prisma/client/runtime/library.js:112:2770)
[ERROR] 21:18:14 PrismaClientInitializationError: The provided database string is invalid. Error parsing connection string: prisma+postgres is not a supported database URL scheme. in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
```

### Severity

🚨 Critical: Data loss, app crash, security issue

### Reproduction

An error appears on server initialisation.

### Expected vs. Actual Behavior

Successful connection to the database.

### Frequency

Consistently reproducible

### Does this occur in development or production?

Both development and production

### Is this a regression?

Yes, last worked in Prisma 6.10.1, broke in 6.11.0


### Workaround

Downgrading to Prisma 6.10.1 fixes it


### Prisma Schema & Queries

```prisma
datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL_PRISMA")
}

generator client {
    provider = "prisma-client-js"
}

generator pothos {
    provider     = "prisma-pothos-types"
    prismaUtils  = true
    output       = "../pothos.d.ts"
    clientOutput = "@prisma/client"
}
```
```ts
export const prisma = new PrismaClient().$extends(withAccelerate());
```


### Prisma Config

```ts
// Add your `prisma.config.ts`
```


### Logs & Debug Info

```
// Debug logs here
```


### Environment & Setup

- **OS:** macOS 15.5
- **Database:** Prisma Postgres
- **Node.js version:** Bun 1.2.17 (Node 22.15.0)


### Prisma Version

```
prisma                  : 6.11.0
@prisma/client          : 6.11.0
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v22.15.0
TypeScript              : 5.8.3
Query Engine (Node-API) : libquery-engine 9c30299f5a0ea26a96790e13f796dc6094db3173 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
PSL                     : @prisma/prisma-schema-wasm 6.11.0-18.9c30299f5a0ea26a96790e13f796dc6094db3173
Schema Engine           : schema-engine-cli 9c30299f5a0ea26a96790e13f796dc6094db3173 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Default Engines Hash    : 9c30299f5a0ea26a96790e13f796dc6094db3173
Studio                  : 0.511.0
```

--- a/packages/client/src/runtime/core/init/validateEngineInstanceConfig.test.ts
+++ b/packages/client/src/runtime/core/init/validateEngineInstanceConfig.test.ts
@@ -201,7 +201,7 @@ describe('validateEngineInstanceConfig', () => {
       expectTrue(ok)
       expect(diagnostics.errors).toBe(undefined)
       expect(diagnostics.warnings).toMatchInlineSnapshot(`[]`)
-      expect(isUsing.accelerate).toBe(false)
+      expect(isUsing.accelerate).toBe(true)
       expect(isUsing.driverAdapters).toBe(true)
       expect(isUsing.ppg).toBe(true)
     })
@@ -243,7 +243,7 @@ describe('validateEngineInstanceConfig', () => {
           },
         ]
       `)
-      expect(isUsing.accelerate).toBe(false)
+      expect(isUsing.accelerate).toBe(true)
       expect(isUsing.driverAdapters).toBe(false)
       expect(isUsing.ppg).toBe(true)
     })

--- a/packages/client/src/runtime/core/init/validateEngineInstanceConfig.ts
+++ b/packages/client/src/runtime/core/init/validateEngineInstanceConfig.ts
@@ -78,20 +78,22 @@ export function validateEngineInstanceConfig({
     errors.push({ _tag: 'error', value })
   }
 
-  const isUsingPrismaAccelerate = Boolean(url?.startsWith('prisma://')) || !copyEngine
+  const isUsingPrismaAccelerate = Boolean(url?.startsWith('prisma://'))
   const isUsingPrismaPostgres = isPrismaPostgres(url)
   const isUsingDriverAdapters = Boolean(adapter)
-  const isCompatibleWithPrismaAccelerate = isUsingPrismaAccelerate || isUsingPrismaPostgres
+  const isAccelerateUrlScheme = isUsingPrismaAccelerate || isUsingPrismaPostgres
 
-  if (!isUsingDriverAdapters && copyEngine && isCompatibleWithPrismaAccelerate) {
+  if (!isUsingDriverAdapters && copyEngine && isAccelerateUrlScheme) {
     pushWarning([
       'recommend--no-engine',
       'In production, we recommend using `prisma generate --no-engine` (See: `prisma generate --help`)',
     ])
   }
 
+  const isAccelerateConfigured = isAccelerateUrlScheme || !copyEngine
+
   // Note: we're explicitly allowing the `isUsingDriverAdapters && isUsingPrismaPostgres` case to pass through.
-  if (isUsingDriverAdapters && (isCompatibleWithPrismaAccelerate || targetBuildType === 'edge')) {
+  if (isUsingDriverAdapters && (isAccelerateConfigured || targetBuildType === 'edge')) {
     if (targetBuildType === 'edge') {
       pushError([
         `Prisma Client was configured to use the \`adapter\` option but it was imported via its \`/edge\` endpoint.`,
@@ -111,7 +113,7 @@ export function validateEngineInstanceConfig({
   }
 
   const isUsing = {
-    accelerate: isUsingPrismaAccelerate,
+    accelerate: isAccelerateConfigured,
     ppg: isUsingPrismaPostgres,
     driverAdapters: isUsingDriverAdapters,
   }

